osquery/core/system.cpp

```diff
@@ -536,4 +536,31 @@ DropPrivileges::~DropPrivileges() {
   }
 }
 #endif
+
+Status setThreadName(const std::string& name) {
+  int return_code;
+#if defined(__APPLE__)
+  return_code = pthread_setname_np(name.c_str());
+#elif defined(__linux__)
+  return_code = pthread_setname_np(pthread_self(), name.c_str());
+#elif defined(WIN32)
+  // SetThreadDescription is available in builds newer than 1607 of windows 10
+  // and works even if there is no debugger.
+  typedef HRESULT(WINAPI * PFNSetThreadDescription)(HANDLE hThread,
+                                                    PCWSTR lpThreadDescription);
+  auto pfnSetThreadDescription = reinterpret_cast<PFNSetThreadDescription>(
+      GetProcAddress(GetModuleHandleA("Kernel32.dll"), "SetThreadDescription"));
+  if (pfnSetThreadDescription) {
+    std::wstring wideName;
+    wideName.resize(name.size() + 1);
+    swprintf_s(&(wideName.front()), wideName.size(), L"%S", name.c_str());
```

Comment:
nit: Would you mind using the helper function we have for this? `wstringToString`?